scripts/run_macos.sh

```diff
@@ -87,8 +87,39 @@ function deactivate_conda_env() {
     fi
 }
 
+# Function to check if Homebrew is installed
+check_homebrew_installed() {
+    if command -v brew >/dev/null 2>&1; then
+        echo "Homebrew is already installed."
+    else
+        echo "Homebrew is not installed."
+        read -p "Would you like to install Homebrew? (y/n): " choice
+        if [[ "$choice" == "y" || "$choice" == "Y" ]]; then
+            install_homebrew
+        else
+            echo "Homebrew installation skipped."
+        fi
+    fi
+}
+
+# Function to install Homebrew
+install_homebrew() {
+    echo "Installing Homebrew..."
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+    if [ $? -eq 0 ]; then
+        echo "Homebrew installed successfully."
+    else
+        echo "Homebrew installation failed."
+    fi
+}
+
 function install_dependencies() {
     # check if the env is already setup by finding 'kotaemon' in 'pip list'
+
+    echo "Installing system dependencies...."
+    check_homebrew_installed
+    brew install libmagic
```

Comment:
Let's verify the need for libmagic and homebrew

Reply:
Yup, see comment above re `unstructured` system dependencies. As for homebrew, it's fairly standard practice to use it on a Mac as an installer for system packages but I'll leave that decision up to you all!